pkg/virt-launcher/virtwrap/network/cloud-init-network.go

```diff
@@ -0,0 +1,251 @@
+package network
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	"gopkg.in/yaml.v2"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type CloudInitNetworkInterface struct {
+	NetworkType string            `yaml:"type"`
+	Name        string            `yaml:"name,omitempty"`
+	Mac_address string            `yaml:"mac_address,omitempty"`
+	Mtu         uint16            `yaml:"mtu,omitempty"`
+	Subnets     []CloudInitSubnet `yaml:"subnets,omitempty"`
+	Address     []string          `yaml:"address,omitempty"`
+	Search      []string          `yaml:"search,omitempty"`
+	Destination string            `yaml:"destination,omitempty"`
+	Gateway     string            `yaml:"gateway,omitempty"`
+	Metric      int               `yaml:"metric,omitempty"`
+}
+
+type CloudInitSubnet struct {
+	SubnetType string                 `yaml:"type"`
+	Address    string                 `yaml:"address"`
+	Gateway    string                 `yaml:"gateway,omitempty"`
+	Routes     []CloudInitSubnetRoute `yaml:"routes,omitempty"`
+}
+
+type CloudInitSubnetRoute struct {
+	Network string `yaml:"network,omitempty"`
+	Netmask string `yaml:"netmask,omitempty"`
+	Gateway string `yaml:"gateway,omitempty"`
+}
+
+type CloudInitConfig struct {
+	Version int                         `yaml:"version"`
+	Config  []CloudInitNetworkInterface `yaml:"config"`
+}
+
+// Borrowed from Convert_v1_VirtualMachine_To_api_Domain
+func getSriovNetworkInfo(vmi *v1.VirtualMachineInstance) ([]VIF, error) {
+	networks := map[string]*v1.Network{}
+	cniNetworks := map[string]int{}
+	var sriovVifs []VIF
+
+	for _, network := range vmi.Spec.Networks {
+		numberOfSources := 0
+		if network.Pod != nil {
+			numberOfSources++
+		}
+		if network.Multus != nil {
+			cniNetworks[network.Name] = len(cniNetworks) + 1
+			numberOfSources++
+		}
+		if network.Genie != nil {
+			cniNetworks[network.Name] = len(cniNetworks)
+			numberOfSources++
+		}
+		if numberOfSources == 0 {
+			return sriovVifs, fmt.Errorf("fail network %s must have a network type", network.Name)
```

Comment:
Hm, I was sure we have all such validation on the webhook level

Reply:
This was copied out of converter. I can remove the test but at the time it seemed prudent to leave it as is.

Reply:
Want me to trim this down?

Reply:
Yes I think you can get rid of it, it already covered by https://github.com/kubevirt/kubevirt/blob/ff476cae19647486941ad63b92597839c02217cb/pkg/virt-api/webhooks/validating-webhook/validating-webhook.go#L852

Reply:
Done.